### PR TITLE
Bugfixes: coordinator event scanning and tdec request/response

### PR DESCRIPTION
**Pull request by derekpierre**
Based over #3093

### CODE REVIEW

File: nucypher/blockchain/eth/trackers/dkg.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def _get_start_block_number(self) -> int:
         ):
             expected_start_block = w3.eth.get_block(expected_start_block.number - 1)
 
-        return expected_start_block.number
```
Comment (derekpierre) [inline nucypher/blockchain/eth/trackers/dkg.py:136]: Do we need to be careful here. Could this return -1 if the expected_start_block.number is 0 - might be possible in tests?

File: nucypher/blockchain/eth/trackers/dkg.py [hunk 2]
```diff
@@ -127,13 +127,20 @@ def _get_start_block_number(self) -> int:
...
         )
-        while expected_start_block.number > 0 and expected_start_block.timestamp > target_timestamp:
+        while (
+            expected_start_block.number > 0
+            and expected_start_block.timestamp > target_timestamp
+        ):
             expected_start_block = w3.eth.get_block(expected_start_block.number - 1)
 
-        return expected_start_block.number
+        return int(expected_start_block.number - 1)
```
Comment (derekpierre) [inline nucypher/blockchain/eth/trackers/dkg.py:143]: Do we need to be careful here. Could this return -1 if the expected_start_block.number is 0 - might be possible in tests?

```suggestion
        return int(expected_start_block.number - 1) if expected_start_block.number > 0 else 0
```

Comment (derekpierre) [reply nucypher/blockchain/eth/trackers/dkg.py:143]: Or maybe adjust the earlier logic to already accomodate the -1...

Comment (derekpierre) [reply nucypher/blockchain/eth/trackers/dkg.py]: Resolved via https://github.com/nucypher/nucypher/pull/3092/commits/958c8ff9968ccd359c2f6f427af6244127c353e8

**Review by KPrasch — APPROVED** [review]

**Pull request merged**